File: accidentes/FicheroAccidentes-Java-IvanRC/src/main/java/mapper/distrito/DistritoMapper.java
package mapper.distrito;

import dto.distrito.DistritoDto;
import model.Distrito;
import utils.Utils;

public class DistritoMapper {

    public static DistritoDto toDto(Distrito distrito){
        return new DistritoDto(
                Utils.getValorNumericoToString(distrito.getId()),
                distrito.getNombre()
    );
    }

    public static Distrito toDistrito(DistritoDto distritoDto){
        return new Distrito(
                Utils.valorNumericoToInt(distritoDto.getId()),
                distritoDto.getNombre()
    );
    }

}

File: accidentes/FicheroAccidentes-Kotlin-IvanRC/src/main/kotlin/mapper/distrito/DistritoMapper.kt
package mapper.distrito

import dto.distrito.DistritoDto
import model.Distrito

fun Distrito.toDto(): DistritoDto{
    return DistritoDto(
        id = if(this.id != null) this.id.toString() else "NULL",
        nombre = this.nombre
    )
}

fun DistritoDto.toDistrito(): Distrito{
    return Distrito(
        id = if(this.id.matches(Regex("[0-9]+"))) this.id.toInt() else null,
        nombre = this.nombre
    )
}